src/System.CommandLine/StringBuilderPool.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.CommandLine
+{
+    public class StringBuilderPool
```

Comment:
This should probably be internal.

Reply:
Should I repeat it across all projects that need it? Or àdd `InternalsVisibleToAttribute`?

Reply:
@jonsequitur, I was about to make `StringBuilderPool` internal and add to the other projects as a link, but that defeats the whole purpose of the shared pool.

Reply:
I typically avoid `InternalsVisibleToAttribute` because the impact of it tends to spread out over time.

We've used linked files in a few places but because this one is stateful, that approach might defeat the purpose.

Because of the lower level of maturity for the other libraries, maybe we should skip this optimization for now? Other ideas?

Reply:
Leave it public?

I understand the concern of leaking a API that is not the concern of the package.

I just think that using [`ObjectPoolProviderExtensions`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.objectpool.objectpoolproviderextensions) might be overkill because of the lifetime of the of the pool and pooled objects.

Relying in an hypothetical global `StringBuilder` pool would be even worst.

Reply:
I don't think it should be public because then people will use it, and we'll have to maintain it. I also don't think we should add dependencies. 

It's unclear if `System.CommandLine.DragonFruit` or `System.CommandLine.Rendering` will reach a GA in their current form so I'm less concerned with optimizing their performance than I am with making sure the `System.CommandLine` surface area is clean.

Reply:
What do you think of taking a dependency on [Microsoft.Extensions.ObjectPool](https://www.nuget.org/packages/Microsoft.Extensions.ObjectPool/)?

I could keep it as a pooled weak references.

Reply:
I'd like to avoid adding a dependency.